### ISSUE TITLE
Radio buttons: .btn-group-toggle: Keyboard selection only works visually.

### ISSUE THREAD

**Issue by He3lixxx** [twbs/bootstrap#26855]
In a toggle button group, if you select a button using the keyboard, the underlying selection of the radio inputs is not changed. Thus, if you have these buttons in a form and submit the form, the user might submit an unintended selection.

I guess this is related to #18874. I added a comment there, but there was no reaction.
#25122 and #26778 are possibly related but no duplicates imo.

### Steps to reproduce:
- In a toggle button group, click on one button with the mouse
- Then, change your selection with the arrow keys (left / right)

### Expected Result:
The radio input that represents the now selected button is checked

### Actual Result:
The radio input selection is not changed although the visuals indicate that it was changed. (tested in firefox 60.0.2, firefox 61.0.1 and chrome 67.0.3396.99)

### Example Code
[You can reproduce the issue here](https://jsbin.com/xatikicere/edit?html,css,output). The html code is copied from [the docs](https://getbootstrap.com/docs/4.1/components/buttons/#checkbox-and-radio-buttons). I only added css to remove the clip property on the inputs so you can see which input is selected.

**Comment by Johann-S** [comment]
you're right it seems the normal behavior of a radio button 🤔 

/CC @patrickhlauke

**Comment by RinkAttendant6** [comment]
I can reproduce this as well on Firefox 61 and Edge 17.17134 

However it seems to work perfectly fine in IE 11 :thinking:

**Comment by NathanCH** [comment]
Can confirm this is an issue in Firefox 64, and Chrome 71, Safari 12.0

**Comment by He3lixxx** [comment]
If someone comes by here and is searching for a solution: This is the work around we came up with: https://github.com/fsr-de/EvaP/pull/1326/files

```javascript
$(document).on("click", '[data-toggle^="button"] input[type="radio"]', function(event) {
  // Bootstrap will call event.preventDefault() which results into a reset of the radio buttons before the keyboard input.
  // To work around this, we set the method to one without side effects.
  event.preventDefault = () => true;
});
```

**Comment by Lausselloic** [comment]
Fix is in progress for V4 and V5 thanks to @patrickhlauke 
https://github.com/twbs/bootstrap/pull/28565

**Comment by patrickhlauke** [comment]
@mdo so the question is: do we want to keep supporting the case where `<button>` elements behave like checkboxes/radio buttons (we seem to have no examples per se in the docs, just a mention in https://getbootstrap.com/docs/4.3/components/button-group/#basic-example and a related unit test which this PR fails (`button should toggle aria-pressed on buttons with container`). i guess if we stopped supporting this, it would be a breaking change, so no chance for a 4.x release?

if we do want to keep supporting this case, the functionality will need to be put back into this PR's javascript...

(no idea why the other unit test `button should check for closest matching toggle` fails at the moment...if the headless browser behaves like a normal browser, it should correctly set the properties for checked etc...will investigate this aspect)

**Comment by patrickhlauke** [comment]
sorry, late reply...fixed with this, i believe https://github.com/twbs/bootstrap/pull/28834

**Comment by LinqLover** [comment]
Are you really sure this has been fixed in v4? For me, the [radio group example from the docs](https://getbootstrap.com/docs/4.0/components/buttons/#checkbox-and-radio-buttons) still does not register the selection in the model if I use arrow keys ...

**Comment by patrickhlauke** [comment]
there were further changes that broke it apparently, but then further fixes after that. in the end the complete fix came in 4.4 https://getbootstrap.com/docs/4.4/components/buttons/#checkbox-and-radio-buttons

in general, suggest updating to the latest version
